src/core/Akka.Persistence/Persistence.cs

```diff
@@ -21,18 +20,21 @@ namespace Akka.Persistence
 {
     internal struct PluginHolder
     {
-        public PluginHolder(IActorRef @ref, EventAdapters adapters, Config config)
+        public PluginHolder(IActorRef @ref, EventAdapters adapters, Config config, IActorRef recoveryPermitter)
         {
             Ref = @ref;
             Adapters = adapters;
             Config = config;
+            RecoveryPermitter = recoveryPermitter;
```

Comment:
Recovery permitter is also cached inside the `PluginHolder` so that it can be looked up.